libyul/YulStack.cpp

```diff
@@ -231,7 +238,15 @@ YulStack::assembleEVMWithDeployed(optional<string_view> _deployName) const
 
 	evmasm::Assembly assembly(m_evmVersion, true, {});
 	EthAssemblyAdapter adapter(assembly);
-	compileEVM(adapter, m_optimiserSettings.optimizeStackAllocation);
+
+	// NOTE: We always need stack optimization when Yul optimizer is disabled (unless code contains
+	// msize). It being disabled just means that we don't use the full step sequence. We still run
+	// it with the minimal steps required to avoid "stack too deep".
+	bool optimize = m_optimiserSettings.optimizeStackAllocation || (
```

Comment:
Can `optimizeStackAllocation` be toggled via the standard JSON interface, and if yes, should tests be added for it, i.e. to make sure that `optimize` is deduced correctly based on the input flags?

Reply:
Yes it can (via `settings.optimizer.details.yulDetails.stackAllocation`) but only when Yul optimizer is officially enabled as well.

Unfortunately this is not very easy to test. This value is not exposed directly so I cannot unit-test it. I could make a command-line test, but I don't like that too much because you can't easily tell just by looking at asm output if the setting was applied or not.

An indirect way to tell that it works in this PR is by the fact that this change did not cause any changes in the output of tests that use `msize` (if it did, I'd include them in the same commit). I actually did get that wrong initially and I was seeing such changes here. They're gone now.

This only covers the `optimizeStackAllocation = true` case. It's possible that we'll soon make this setting permanently `true` anyway, so it's not worth putting too much time into testing the other case IMO.